### PR TITLE
Compilation works when we enable backend optimisation in `-O0` optimisation level.

### PR DESCRIPTION
**Pull request by nmdis1999**
Currently if we try to remove optnone from backend optimisation yklua's compilation fails with error:`Unrecognzed operand type`. This error occurs because we hadn't handle `ConstantOp` case in `FixStackmapSpillsReloads.cpp`. After completing the case for `ConstantOp` and using the new build to compile yklua we can fix this issue. I have added reduced test case as well in this PR.

### CODE REVIEW

**Comment by nmdis1999** [comment]
@ltratt @ptersilie ready for review.

**Comment by ltratt** [comment]
Can we change the PR title to something like (please reword this as appropriate!) "Handle ConstantOp in ..." because otherwise it's a bit difficult to know what the PR is doing specifically.

**Comment by nmdis1999** [comment]
Done.

File: llvm/test/CodeGen/X86/yk-o0-fixstackmap-bug.ll [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+!94 = !DILocation(line: 27, column: 3, scope: !78)
+!95 = distinct !{!95, !80, !96, !97}
+!96 = !DILocation(line: 32, column: 3, scope: !74)
+!97 = !{!"llvm.loop.mustprogress"}
+!98 = !DILocation(line: 34, column: 3, scope: !38)
+!99 = !DILocation(line: 35, column: 14, scope: !38)
+!100 = !DILocation(line: 35, column: 3, scope: !38)
+!101 = !DILocation(line: 37, column: 3, scope: !38)
+
+^0 = module: (path: "[Regular LTO]", hash: (0, 0, 0, 0, 0))
```
Comment (ptersilie) [inline llvm/test/CodeGen/X86/yk-o0-fixstackmap-bug.ll:238]: Is this stuff required for the test to work? I've never seen this before in any module dump.

**Comment by ptersilie** [comment]
I think we should fix the commit title as well to something like
```
Fix handling of constants in FixStackmapSpillReloads pass.
```

File: llvm/test/CodeGen/X86/yk-o0-fixstackmap-bug.ll (hunk 1 above)
Comment (nmdis1999) [reply llvm/test/CodeGen/X86/yk-o0-fixstackmap-bug.ll:238]: yes, if I remove this the test fails.

**Comment by ltratt** [comment]
Is this the minimal `.ll` file that causes a problem? Has it been run through `llvm-reduce`?

**Comment by nmdis1999** [comment]
yes, I haven't used llvm-reduce is it same as creduce?

**Comment by ltratt** [comment]
Please have a look at llvm-reduce and see if it helps.

File: llvm/test/CodeGen/X86/yk-o0-fixstackmap-bug.ll (hunk 1 above)
Comment (vext01) [reply llvm/test/CodeGen/X86/yk-o0-fixstackmap-bug.ll:238]: I've also never seen this syntax!

Comment (vext01) [reply llvm/test/CodeGen/X86/yk-o0-fixstackmap-bug.ll:238]: https://llvm.org/docs/LangRef.html#module-path-summary-entry

It's to do with Thin LTO, where you use module summaries. We don't use thin LTO, but I guess it bleeds across a bit somehow.

**Comment by vext01** [comment]
I'm about 80% sure you can remove all the debug metadata stuff from the module too? (`!` stuff).

**Comment by nmdis1999** [comment]
So the problem was I generated `.ll` file with yk flags too when I don't think they were needed so we are seeing `LTO` stuff in the IR. Fixed that.

**Comment by ptersilie** [comment]
That looks much better. Please squash (and rename the commit message).

**Comment by nmdis1999** [comment]
Squashed.

**Comment by ltratt** [comment]
Before we merge, @ptersilie I think the test file should have a better name than "fixstackmapbug"?

**Comment by ptersilie** [comment]
Yes, I missed that. How about "yk-stackmaps-spillreloads-fix-consts.ll`?

**Comment by nmdis1999** [comment]
@ptersilie done, please lmk if i should squash?

**Comment by ptersilie** [comment]
Please squash.

**Comment by nmdis1999** [comment]
Squashed.

**Comment by ltratt** [comment]
Please force push an update to the commit message explaining what the commit is doing (the title isn't enough).

**Comment by nmdis1999** [comment]
@ltratt please lmk if the commit message make sense now?
